applications/mne_scan/plugins/eegosports/FormFiles/eegosportsimpedancewidget.cpp

```diff
@@ -0,0 +1,391 @@
+            svgGen.setSize(QSize(rect.width(), rect.height()));
+            //svgGen.setViewBox(QRect(0, 0, rect.width(), rect.height()));
+
+            QPainter painter(&svgGen);
+            m_qGScene->render(&painter);
+        }
+
+        if(fileName.contains(".png"))
+        {
+            m_qGScene->setSceneRect(m_qGScene->itemsBoundingRect());                          // Re-shrink the scene to it's bounding contents
+            QImage image(m_qGScene->sceneRect().size().toSize(), QImage::Format_ARGB32);  // Create the image with the exact size of the shrunk scene
+            image.fill(Qt::transparent);                                              // Start all pixels transparent
+
+            QPainter painter(&image);
+            m_qGScene->render(&painter);
+            image.save(fileName);
+        }
+    }
+}
+
+//*************************************************************************************************************
+
+void EEGoSportsImpedanceWidget::loadLayout()
+{
+    QString sElcFilePath = QFileDialog::getOpenFileName(this,
+                                                        tr("Open Layout"),
+                                                        "./resources/mne_scan/plugins/eegosports/loc_files/",
+                                                        tr("ELC layout file (*.elc)"));
+
+    // Load standard layout file
+//    LayoutLoader *asaObject = new LayoutLoader();
+    QList<QVector<float> > elcLocation3D;
+    QList<QVector<float> > elcLocation2D;
+    QString unit;
+    QStringList elcChannelNames;
+
+    if(!LayoutLoader::readAsaElcFile(sElcFilePath, elcChannelNames, elcLocation3D, elcLocation2D, unit))
+        qDebug() << "Error: Reading elc file.";
+    else
+        m_qGScene->clear();
+
+    // Clean old map -> Generate lookup table for channel names and corresponding index
+    m_qmElectrodeNameIndex.clear();
+
+    for(int i = 0; i<elcLocation2D.size(); i++)
+        m_qmElectrodeNameIndex.insert(elcChannelNames.at(i), i);
+
+    // Add electrodes to scene
+    for(int i = 0; i<elcLocation2D.size(); i++)
+    {
+        QVector2D position(elcLocation2D[i][1]*-4.5,elcLocation2D[i][0]*-4.5); // swap x y to rotate 90°, multiply to mirror by x and y axis, multiply by to scale
+
+        addElectrodeItem(elcChannelNames.at(i), position);
+    }
+
+    ui->m_graphicsView_impedanceView->fitInView(m_qGScene->itemsBoundingRect(), Qt::KeepAspectRatio);
+}
+
+//*************************************************************************************************************
+
+void EEGoSportsImpedanceWidget::closeEvent(QCloseEvent *event)
+{
+    Q_UNUSED(event);
+
+    // On window close event -> stop impedance measurement
+    if(m_pEEGoSports->m_bIsRunning)
+        stopImpedanceMeasurement();
+}
+
+//*************************************************************************************************************
+
+// This function is needed to sort the QList
```

Comment:
please add comments to the header and declaration of the function.

Reply:
This is a carry-over from tmsiimpedancewidget.cpp. I find this a bit odd anyway, since the function is not a member-function and not declared in the header-file. Should this stay as is, only that separator comments are added?

Reply:
ah I guess I defined it as a global function so we can use `std::sort`. Can you try defining this function as a static member function in `EEGoSportsImpedanceWidget`?

Reply:
It is defined as a private static member function now, which doesn't seem to be a problem for `std::sort`